lib/websql/WebSQLDatabase.js

```diff
@@ -22,12 +38,54 @@ function TransactionTask(readOnly, txnCallback, errorCallback, successCallback)
   this.successCallback = successCallback;
 }
 
-function WebSQLDatabase(dbVersion, db) {
-  this.version = dbVersion;
-  this._db = db;
-  this._txnQueue = new Queue();
-  this._running = false;
-  this._currentTask = null;
+function WebSQLDatabase(dbVersion, db, callback) {
+  var self = this;
+  self.version = dbVersion;
+  self._db = db;
+  self._txnQueue = new Queue();
+  self._running = false;
+  self._currentTask = null;
+  self._callback = callback;
+
+  self._db.exec([createInfoTable], false, function (error) {
+    /* istanbul ignore next */
+    if (error) {
+      return self._callback(null, error);
+    }
+    self._db.exec([getVersion], true, function (error, res) {
+      /* istanbul ignore next */
+      if (error) {
+        return self._callback(null, error);
+      }
+      if (res[0].rows.length === 1) {
+        var currentVersion = res[0].rows[0].value;
+        if (self.version !== '' && self.version !== currentVersion) {
+          return self._callback(null, {
+            code: 11,
+            message: 'unable to open database, version mismatch, \'' + 
+              self.version + '\' does not match the currentVersion of \'' +
+              currentVersion + '\'',
+            name: 'InvalidStateError'
+          });
+        } else {
+          self.version = currentVersion;
+          self._callback(self);
+        }
+      } else {
+        var setVer = setVersion;
+        setVer.args = [self.version];
+        
+        self._db.exec([setVer], false, function (error) {
+          /* istanbul ignore if */
+          if (error) {
+            return self._callback(null, error);
+          } else {
+            self._callback(self);
+          }
```

Comment:
```suggestion
          
          self._callback(self);          
```

The `else` is unnecessary here because of the `return`